internal/testutil/testutil.go

```diff
@@ -89,24 +90,45 @@ func diffSet(path string) sets.String {
 // may have been changed after the destDir was copied, it is often better to explicitly
 // use a set of golden files as the sourceDir rather than the original TestGitRepo
 // that was copied.
-func (g *TestGitRepo) AssertEqual(t *testing.T, sourceDir, destDir string) bool {
-	return AssertPkgEqual(t, sourceDir, destDir)
-}
-
-func AssertPkgEqual(t *testing.T, sourceDir, destDir string) bool {
-	diff, err := Diff(sourceDir, destDir)
+func (g *TestGitRepo) AssertEqual(t *testing.T, sourceDir, destDir string, addMergeCommentsToSource bool) bool {
+	diff, err := Diff(sourceDir, destDir, addMergeCommentsToSource)
 	if !assert.NoError(t, err) {
 		return false
 	}
 	diff = diff.Difference(KptfileSet)
 	return assert.Empty(t, diff.List())
 }
 
+// addMergeCommentToExpected copies the expected path directory contents to
+// new temp directory and adds merge comment to the resources in directory
+// it also returns the cleanup function to clean the created temp directory
+func addMergeCommentToExpected(path string) (string, func(), error) {
```

Comment:
I find this name confusing. What this has to do with `expected` is unclear.

Reply:
My bad. Moved it to addmergecomments package itself. It is cleaner now.